### ISSUE TITLE
Consensus sequence

### ISSUE THREAD

**Issue by MrBleem** [DrosophilaGenomeEvolution/TrEMOLO#25]
Hi, 

I just running the OUTSIDER part, and I want to get the consensus sequence or insertion sequence for each insertions.
Could you tell me which file in the ouput contains the sequence information?

Best 
Bo

**Comment by M-D75** [comment]
Hi,

You can find the sequences associated with your TEs in the following files:

* OUTSIDER/TrEMOLO_SV_TE/INS/SV_INS_CLUST.fasta (for TrEMOLO IDs)
* OUTSIDER/VARIANT_CALLING/SEQUENCE_INDEL.fasta (for sniffles IDs).

The TE identifiers are of two types: TrEMOLO.INS.[0-9]+ and sniffles.INS.[0-9]+.

In the fasta files, the sequence headers look like this:

```
>chrX:<INS>:22714472:22714473:TrEMOLO.INS.84:36:IMPRECISE:33
```
The last number (33) corresponds to the read support number for the corresponding TE.

Here’s an example of a bash command to retrieve them while filtering out unnecessary sequences:

```bash
cut -f 4 /path/to/your_workdirectory/TE_INFOS.bed | cut -d "|" -f 2 > ID.txt

grep -w -f ID.txt /path/to/your_workdirectory/OUTSIDER/VARIANT_CALLING/SEQUENCE_INDEL.fasta > READ_SEQUENCE.fasta

grep -w -f ID.txt /path/to/your_workdirectory/OUTSIDER/TrEMOLO_SV_TE/INS/SV_INS_CLUST.fasta >> READ_SEQUENCE.fasta
```

Feel free to let me know if anything is unclear.

Best,
M-D

**Comment by MrBleem** [comment]
The sequences in SEQUENCE_INDEL.fasta contains are duplicates, can I use the the OUTSIDER/ET_FIND_FA/${TE}.fasta

**Comment by MrBleem** [comment]
Hi,
The sequences in SEQUENCE_INDEL.fasta contains are duplicated, can I use the the OUTSIDER/ET_FIND_FA/${TE}.fasta?

**Comment by M-D75** [comment]
Yes, indeed, it is actually better to use the sequences contained in **ET_FIND_FA**. However, please note that these sequences are slightly truncated, and in some cases, they might have been shortened too much.



> By the way, I don't find any TrEMOLO IDs in TE_INFOS.bed, maybe because I just run the OUTSIDER part, or something else?


Indeed, it is the `OUTSIDER` part that is supposed to provide you with TEs with this type of ID. However, it is possible that the TrEMOLO identification process has not found anything more than sniffles (sniffles takes priority).

**Comment by MrBleem** [comment]
ok, get it! Thank you very much!

**Comment by MrBleem** [comment]
Oh, one more question, do you make a local assembly process for each insertion, or just provide one of the supporting reads?

**Comment by M-D75** [comment]
No, we do not create a consensus, we simply take the most relevant sequence from the read support. If I understand correctly, you do not want to create a consensus for all your TEs by family, but rather a consensus for each TE insertion, using the entire set of read supports for each insertion ? 

if that's the case, then you should go with option number one for sequence retrieval, with some cleaning to do

**Comment by MrBleem** [comment]
> a consensus for each TE insertion 
Yeah, that's it.

OK, I will try it.
